src/devices/Ft4222/Ft4222I2c.cs

```diff
@@ -46,98 +56,59 @@ public Ft4222I2c(I2cConnectionSettings settings)
+        /// <inheritdoc/>
         public override byte ReadByte()
         {
-            byte[] toRead = new byte[1];
+            Span<byte> toRead = stackalloc byte[1];
             Read(toRead);
             return toRead[0];
         }
 
-        /// <summary>
-        /// Writes data to the I2C device.
-        /// </summary>
-        /// <param name="buffer">
-        /// The buffer that contains the data to be written to the I2C device.
-        /// The data should not include the I2C device address.
-        /// </param>
+        /// <inheritdoc/>
         public override void Write(ReadOnlySpan<byte> buffer)
         {
-            ushort byteSent = 0;
-            var ftStatus = FtFunction.FT4222_I2CMaster_Write(_ftHandle, (ushort)_settings.DeviceAddress, buffer.ToArray(), (ushort)buffer.Length, ref byteSent);
+            ushort byteSent;
+            var ftStatus = FtFunction.FT4222_I2CMaster_Write(_ftHandle, (ushort)_settings.DeviceAddress, in MemoryMarshal.GetReference(buffer), (ushort)buffer.Length, out byteSent);
             if (ftStatus != FtStatus.Ok)
                 throw new IOException($"{nameof(Write)} failed to write, error: {ftStatus}");
         }
 
-        /// <summary>
-        /// Writes a byte to the I2C device.
-        /// </summary>
-        /// <param name="value">The byte to be written to the I2C device.</param>
+        /// <inheritdoc/>
         public override void WriteByte(byte value)
         {
-            Write(new byte[] { value });
+            Span<byte> toWrite = stackalloc byte[1] { value };
+            Write(toWrite);
         }
 
-        /// <summary>
-        /// Performs an atomic operation to write data to and then read data from the I2C bus on which the device is connected, 
-        /// and sends a restart condition between the write and read operations.
-        /// </summary>
-        /// <param name="writeBuffer">
-        /// The buffer that contains the data to be written to the I2C device.
-        /// The data should not include the I2C device address.</param>
-        /// <param name="readBuffer">
-        /// The buffer to read the data from the I2C device.
-        /// The length of the buffer determines how much data to read from the I2C device.
-        /// </param>
+        /// <inheritdoc/>
         public override void WriteRead(ReadOnlySpan<byte> writeBuffer, Span<byte> readBuffer)
         {
             Write(writeBuffer);
             Read(readBuffer);
         }
 
-        /// <summary>
-        /// Dispose the class
-        /// </summary>
-        /// <param name="disposing"></param>
+        /// <inheritdoc/>
         protected override void Dispose(bool disposing)
         {
-            if (_ftHandle != IntPtr.Zero)
-            {
-                FtFunction.FT4222_UnInitialize(_ftHandle);
-                FtFunction.FT_Close(DeviceInformation.FtHandle);
-            }
+            if (!_ftHandle.IsClosed)
```

Comment:
I think if you just call Dispose this exact logic will happen internally:
https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.cs#L174